DA/Internal/Desugar.hs

```diff
@@ -34,6 +34,7 @@ data Text
 data Optional a = None | Some a
 
 data Consuming t = Consuming {}
+data NonConsuming t = NonConsuming {}
```

Comment:
not really related, but I kept adding this for my tests so I thought I'd just add it upstream

Reply:
definitely makes sense, surprised it took us so long to add it